integration/test_collection_model.py

```diff
@@ -407,3 +408,21 @@ class TestRefPropUpdate(BaseProperty):
 
     create_original_collection()
     client.collection_model.update(TestRefPropUpdate)
+
+
+def test_model_with_datetime_property(client: weaviate.Client):
+    class TestDatetime(BaseProperty):
+        name: str
+        date: datetime
+
+    client.collection_model.delete(TestDatetime)
+    collection = client.collection_model.create(
+        CollectionModelConfig[TestDatetime](model=TestDatetime, vectorizer=Vectorizer.NONE)
+    )
+    now = datetime.now(timezone.utc)
+    collection.data.insert(TestDatetime(name="test", date=now))
+    objects = collection.data.get()
+    assert len(objects) == 1
+    assert objects[0].data.name == "test"
+    assert type(objects[0].data.date) is datetime
+    # assert objects[0].data.date == now
```

Comment:
what is the problem here?

Reply:
The issue of comparing a Python defined datetime, which has 5 microseconds, and the Weaviate saved date, which has 3 microseconds and 2 trailing zeros due to the rounding up to 3 sig fig. This needs the changes added to it that we discussed around fuzzy comparison, which I will do now 😁

Reply:
Aha, the truncation was actually my own mistake mixing up milliseconds and microseconds 😅 so it's all fine!

Reply:
if you want to leave the commented out code - could you add why it is currently commented out?

Reply:
It shouldn't be commented out as it ought to work now, will test it out